crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2575,7 +2575,7 @@ pub async fn list_payment_methods(
     let merchant_surcharge_configs =
         if let Some((payment_attempt, payment_intent, business_profile)) = payment_attempt
             .as_ref()
-            .zip(payment_intent)
+            .zip(payment_intent.clone())
```

Comment:
We can avoid this clone in paymen_intent if `payment_intent. request_external_three_ds_authentication` is cloned.